core/datatrak.c

```diff
@@ -588,10 +588,9 @@ static int wlog_header_parser (struct memblock *mem)
 	unsigned char *runner = (unsigned char *) mem->buffer;
 	if (!runner)
 		return -1;
-	tmp = (runner[1] << 8) + runner[0];
-	if (tmp == 0x0252) {
+	if (!memcmp(runner, "\x52\x02", 2)) {
 		runner += 8;
-		tmp = (runner[1] << 8) + runner[0];
+		tmp = (int) two_bytes_to_int(runner[1], runner[0]);
```

Comment:
signed vs unsigned integer math is always something that makes me nervous. If someone for some stupid reason has a dive number 33000 (and that is a LOT of diving), this will do weird things. But... yeah.
I would prefer to have an error check for this situation, TBH, but we have many other spots with similar issues. And I consider this an exceptionally rare use case for Subsurface in the first place.